sympy/functions/special/bessel.py

```diff
@@ -1124,9 +1124,11 @@ def fdiff(self, argindex=1):
         else:
             raise ArgumentIndexError(self, argindex)
 
-    @staticmethod
```

Comment:
I'm not clear why talor_term is a staticmethod, but does this break the API?

Reply:
@asmeurer being a staticmethod implies `sin(2*x).taylor_term(1, x)` and `sin(x).taylor_term(1, x)` will return the same thing, which doesn't seem right. Of course, removing the decorator is not the only thing to do here.

Reply:
Yes, I don't understand why it is done that way, but I noticed that all the other taylor_term methods in SymPy are staticmethods.

Reply:
This is the only place I found in the code where `taylor_term` is called on a class, and it can just be replaced with `self.taylor_term`: https://github.com/sympy/sympy/blob/61e5c20c02328815270ddda385f0108a2b40d24d/sympy/functions/elementary/exponential.py#L766

I think we can most likely remove the static method. If someone needs to generate the taylor term for specific class they can create an instance with a dummy variable. It's a (minor) backwards compatibility break so if we do refactor this we should be sure to mention it in the relevant section of the release notes.